checker.c

```diff
@@ -1,18 +1,48 @@
 #include <stdio.h>
 #include <assert.h>
+#include "batteryManagement.h"
 
-int batteryIsOk(float temperature, float soc, float chargeRate) {
+int checkChargeRate(float chargeRate)
+{
+   if(chargeRate > 0.8) {
+    printf("Charge Rate out of range!\n");
+    return 0;
+  }
+  else{
+    return 1;
+  }
+}
+int checkTemperature(float temperature)
+{
   if(temperature < 0 || temperature > 45) {
```

Comment:
Similar range checking duplicate code can be reduced and refactored to a single method. for ex. IsValueInRange(...)